// node-myanimelist/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var anime_1 = require("./methods/jikan/anime");
var manga_1 = require("./methods/jikan/manga");
var person_1 = require("./methods/jikan/person");
var character_1 = require("./methods/jikan/character");
var search_1 = require("./methods/jikan/search");
var season_1 = require("./methods/jikan/season");
var seasonArchive_1 = require("./methods/jikan/seasonArchive");
var seasonLater_1 = require("./methods/jikan/seasonLater");
var schedule_1 = require("./methods/jikan/schedule");
var top_1 = require("./methods/jikan/top");
var genre_1 = require("./methods/jikan/genre");
var producer_1 = require("./methods/jikan/producer");
var magazine_1 = require("./methods/jikan/magazine");
var user_1 = require("./methods/jikan/user");
var club_1 = require("./methods/jikan/club");
// Types
var animeGenre_1 = require("./methods/jikan/types/animeGenre");
var mangaGenre_1 = require("./methods/jikan/types/mangaGenre");
var animeType_1 = require("./methods/jikan/types/animeType");
var mangaType_1 = require("./methods/jikan/types/mangaType");
var animeStatus_1 = require("./methods/jikan/types/animeStatus");
var mangaStatus_1 = require("./methods/jikan/types/mangaStatus");
var animeOrderBy_1 = require("./methods/jikan/types/animeOrderBy");
var mangaOrderBy_1 = require("./methods/jikan/types/mangaOrderBy");
var sort_1 = require("./methods/jikan/types/sort");
var rating_1 = require("./methods/jikan/types/rating");
//
var JikanClient = /** @class */ (function () {
    function JikanClient() {
        //Jikan API
        this.anime = anime_1.default;
        this.manga = manga_1.default;
        this.person = person_1.default;
        this.character = character_1.default;
        this.search = search_1.default;
        this.season = season_1.default;
        this.seasonArchive = seasonArchive_1.default;
        this.seasonLater = seasonLater_1.default;
        this.schedule = schedule_1.default;
        this.top = top_1.default;
        this.genre = genre_1.default;
        this.producer = producer_1.default;
        this.magazine = magazine_1.default;
        this.user = user_1.default;
        this.club = club_1.default;
        this.types = {
            AnimeGenre: animeGenre_1.default,
            MangaGenre: mangaGenre_1.default,
            AnimeType: animeType_1.default,
            MangaType: mangaType_1.default,
            AnimeStatus: animeStatus_1.default,
            MangaStatus: mangaStatus_1.default,
            AnimeOrderBy: animeOrderBy_1.default,
            MangaOrderBy: mangaOrderBy_1.default,
            Sort: sort_1.default,
            Rating: rating_1.default
        };
    }
    return JikanClient;
}());
exports.JikanClient = JikanClient;
var noApiLogin_1 = require("./methods/poly/noApiLogin");
var notifications_1 = require("./methods/poly/loginRequired/notifications");
var animeEdit_1 = require("./methods/poly/loginRequired/animeEdit");
var ScraperClient = /** @class */ (function () {
    function ScraperClient() {
        this.login = noApiLogin_1.default;
        this.notifications = notifications_1.default;
        this.animeEdit = animeEdit_1.default;
    }
    return ScraperClient;
}());
exports.ScraperClient = ScraperClient;
var Mal = new JikanClient();
exports.Mal = Mal;
var version = require("../package.json").version;
exports.version = version;
exports.default = Mal;
