// node-myanimelist/dist/methods/mobileApi/auth/refreshToken.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var axios_1 = require("axios");
var mobileApis_1 = require("../mobileApis");
var qs = function (obj) {
    return Object.keys(obj)
        .map(function (key) { return encodeURIComponent(key) + "=" + encodeURIComponent(obj[key]); })
        .join("&");
};
function default_1(refresh_token, grant_type, client_id) {
    if (grant_type === void 0) { grant_type = "refresh_token"; }
    if (client_id === void 0) { client_id = "6114d00ca681b7701d1e15fe11a4987e"; }
    var request = {
        method: "post",
        url: mobileApis_1.secondaryApiUrl + "/oauth2/token",
        headers: { "content-type": "application/x-www-form-urlencoded" },
        data: qs({
            client_id: client_id,
            refresh_token: refresh_token,
            grant_type: grant_type
        })
    };
    return new Promise(function (res, rej) {
        axios_1.default(request)
            .then(function (r) { return res(r.data); })
            .catch(function (err) { return rej(err); });
    });
}
exports.default = default_1;
