// node-myanimelist/dist/methods/jikan/anime.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var url_1 = require("./url");
var jikanApi_1 = require("./jikanApi");
var axios_1 = require("axios");
var Anime = /** @class */ (function () {
    function Anime(id) {
        this.baseUrl = jikanApi_1.default + "/anime/" + id;
    }
    Anime.prototype.jikanGet = function (url) {
        return axios_1.default.get(url);
    };
    Anime.prototype.info = function () {
        return this.jikanGet(this.baseUrl);
    };
    Anime.prototype.charactersStaff = function () {
        return this.jikanGet(url_1.joinUrl(this.baseUrl, ["characters_staff"]));
    };
    Anime.prototype.episodes = function (p) {
        var params = ["episodes"];
        if (p != null)
            params.push(String(p));
        return this.jikanGet(url_1.joinUrl(this.baseUrl, params));
    };
    Anime.prototype.news = function () {
        return this.jikanGet(url_1.joinUrl(this.baseUrl, ["news"]));
    };
    Anime.prototype.pictures = function () {
        return this.jikanGet(url_1.joinUrl(this.baseUrl, ["pictures"]));
    };
    Anime.prototype.videos = function () {
        return this.jikanGet(url_1.joinUrl(this.baseUrl, ["videos"]));
    };
    Anime.prototype.stats = function () {
        return this.jikanGet(url_1.joinUrl(this.baseUrl, ["stats"]));
    };
    Anime.prototype.forum = function () {
        return this.jikanGet(url_1.joinUrl(this.baseUrl, ["forum"]));
    };
    Anime.prototype.moreInfo = function () {
        return this.jikanGet(url_1.joinUrl(this.baseUrl, ["moreinfo"]));
    };
    Anime.prototype.reviews = function (p) {
        var params = ["reviews"];
        if (p != null)
            params.push(String(p));
        return this.jikanGet(url_1.joinUrl(this.baseUrl, params));
    };
    Anime.prototype.recommendations = function () {
        return this.jikanGet(url_1.joinUrl(this.baseUrl, ["recommendations"]));
    };
    Anime.prototype.userUpdates = function (p) {
        var params = ["userupdates"];
        if (p != null)
            params.push(String(p));
        return this.jikanGet(url_1.joinUrl(this.baseUrl, params));
    };
    return Anime;
}());
function default_1(id) {
    return new Anime(id);
}
exports.default = default_1;
