// node-myanimelist/dist/methods/jikan/manga.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var url_1 = require("./url");
var jikanApi_1 = require("./jikanApi");
var axios_1 = require("axios");
var Manga = /** @class */ (function () {
    function Manga(id) {
        this.baseUrl = jikanApi_1.default + "/manga/" + id;
    }
    Manga.prototype.jikanGet = function (url) {
        return axios_1.default.get(url);
    };
    Manga.prototype.info = function () {
        return this.jikanGet(this.baseUrl);
    };
    Manga.prototype.characters = function () {
        return this.jikanGet(url_1.joinUrl(this.baseUrl, ["characters"]));
    };
    Manga.prototype.news = function () {
        return this.jikanGet(url_1.joinUrl(this.baseUrl, ["news"]));
    };
    Manga.prototype.pictures = function () {
        return this.jikanGet(url_1.joinUrl(this.baseUrl, ["pictures"]));
    };
    Manga.prototype.stats = function () {
        return this.jikanGet(url_1.joinUrl(this.baseUrl, ["stats"]));
    };
    Manga.prototype.forum = function () {
        return this.jikanGet(url_1.joinUrl(this.baseUrl, ["forum"]));
    };
    Manga.prototype.moreInfo = function () {
        return this.jikanGet(url_1.joinUrl(this.baseUrl, ["moreinfo"]));
    };
    Manga.prototype.reviews = function (p) {
        var params = ["reviews"];
        if (p != null)
            params.push(String(p));
        return this.jikanGet(url_1.joinUrl(this.baseUrl, params));
    };
    Manga.prototype.recommendations = function () {
        return this.jikanGet(url_1.joinUrl(this.baseUrl, ["recommendations"]));
    };
    Manga.prototype.userUpdates = function (p) {
        var params = ["userupdates"];
        if (p != null)
            params.push(String(p));
        return this.jikanGet(url_1.joinUrl(this.baseUrl, params));
    };
    return Manga;
}());
function default_1(id) {
    return new Manga(id);
}
exports.default = default_1;
