// node-myanimelist/dist/methods/jikan/schedule.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var url_1 = require("./url");
var jikanApi_1 = require("./jikanApi");
var axios_1 = require("axios");
var Schedule = /** @class */ (function () {
    function Schedule() {
        this.baseUrl = jikanApi_1.default + "/schedule";
    }
    Schedule.prototype.jikanGet = function (url) {
        return axios_1.default.get(url);
    };
    Schedule.prototype.all = function () {
        return this.jikanGet(this.baseUrl);
    };
    Schedule.prototype.monday = function () {
        return this.jikanGet(url_1.joinUrl(this.baseUrl, ["monday"]));
    };
    Schedule.prototype.tuesday = function () {
        return this.jikanGet(url_1.joinUrl(this.baseUrl, ["tuesday"]));
    };
    Schedule.prototype.wednesday = function () {
        return this.jikanGet(url_1.joinUrl(this.baseUrl, ["wednesday"]));
    };
    Schedule.prototype.thursday = function () {
        return this.jikanGet(url_1.joinUrl(this.baseUrl, ["thursday"]));
    };
    Schedule.prototype.friday = function () {
        return this.jikanGet(url_1.joinUrl(this.baseUrl, ["friday"]));
    };
    Schedule.prototype.saturday = function () {
        return this.jikanGet(url_1.joinUrl(this.baseUrl, ["saturday"]));
    };
    Schedule.prototype.sunday = function () {
        return this.jikanGet(url_1.joinUrl(this.baseUrl, ["sunday"]));
    };
    Schedule.prototype.other = function () {
        return this.jikanGet(url_1.joinUrl(this.baseUrl, ["other"]));
    };
    Schedule.prototype.unknown = function () {
        return this.jikanGet(url_1.joinUrl(this.baseUrl, ["unknown"]));
    };
    return Schedule;
}());
function default_1() {
    return new Schedule();
}
exports.default = default_1;
