// node-myanimelist/dist/methods/jikan/search.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var url_1 = require("./url");
var jikanApi_1 = require("./jikanApi");
var axios_1 = require("axios");
var Search = /** @class */ (function () {
    function Search() {
        this.baseUrl = jikanApi_1.default + "/search";
    }
    Search.prototype.jikanGet = function (url) {
        return axios_1.default.get(url);
    };
    Search.prototype.search = function (type, params) {
        var qparams = Object.keys(params)
            .filter(function (k) { return params[k] != null; })
            .map(function (k) { return k + "=" + encodeURIComponent(params[k]); })
            .join("&");
        return this.jikanGet(url_1.joinUrl(this.baseUrl, [type, "?" + qparams]));
    };
    Search.prototype.anime = function (params) {
        return this.search("anime", params);
    };
    Search.prototype.manga = function (params) {
        return this.search("manga", params);
    };
    Search.prototype.person = function (params) {
        return this.search("person", params);
    };
    Search.prototype.character = function (params) {
        return this.search("character", params);
    };
    return Search;
}());
function default_1() {
    return new Search();
}
exports.default = default_1;
