// node-myanimelist/dist/methods/jikan/user.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var url_1 = require("./url");
var jikanApi_1 = require("./jikanApi");
var axios_1 = require("axios");
var User = /** @class */ (function () {
    function User(username) {
        this.baseUrl = jikanApi_1.default + "/user/" + username;
    }
    User.prototype.jikanGet = function (url) {
        return axios_1.default.get(url);
    };
    User.prototype.profile = function () {
        return this.jikanGet(url_1.joinUrl(this.baseUrl, ["profile"]));
    };
    User.prototype.history = function () {
        return new UserHistory(this);
    };
    User.prototype.friends = function (p) {
        var params = ["friends"];
        if (p != null)
            params.push(String(p));
        return this.jikanGet(url_1.joinUrl(this.baseUrl, params));
    };
    User.prototype.animelist = function (p) {
        return new UserAnimelist(this, p);
    };
    User.prototype.mangalist = function (p) {
        return new UserMangalist(this, p);
    };
    return User;
}());
var UserHistory = /** @class */ (function () {
    function UserHistory(parent) {
        this.parent = parent;
    }
    UserHistory.prototype.all = function () {
        // @ts-ignore
        var url = url_1.joinUrl(this.parent.baseUrl, ["history"]);
        // @ts-ignore
        return this.parent.jikanGet(url);
    };
    UserHistory.prototype.anime = function () {
        // @ts-ignore
        var url = url_1.joinUrl(this.parent.baseUrl, ["history", "anime"]);
        // @ts-ignore
        return this.parent.jikanGet(url);
    };
    UserHistory.prototype.manga = function () {
        // @ts-ignore
        var url = url_1.joinUrl(this.parent.baseUrl, ["history", "manga"]);
        // @ts-ignore
        return this.parent.jikanGet(url);
    };
    return UserHistory;
}());
var UserAnimelist = /** @class */ (function () {
    function UserAnimelist(parent, p) {
        this.parent = parent;
        if (p != null)
            this.page = String(p);
        else
            this.page = "1";
    }
    UserAnimelist.prototype.getList = function (params, qparams) {
        if (qparams != null) {
            var qparams_1 = Object.keys(params)
                .filter(function (k) { return params[k] != null; })
                .map(function (k) { return k + "=" + encodeURIComponent(params[k]); })
                .join("&");
            params.push(qparams_1);
        }
        // @ts-ignore
        var url = url_1.joinUrl(this.parent.baseUrl, params);
        // @ts-ignore
        return this.parent.jikanGet(url);
    };
    UserAnimelist.prototype.all = function (qparams) {
        return this.getList(["animelist", "all", this.page], qparams);
    };
    UserAnimelist.prototype.watching = function (qparams) {
        return this.getList(["animelist", "watching", this.page], qparams);
    };
    UserAnimelist.prototype.onhold = function (qparams) {
        return this.getList(["animelist", "onhold", this.page], qparams);
    };
    UserAnimelist.prototype.dropped = function (qparams) {
        return this.getList(["animelist", "dropped", this.page], qparams);
    };
    UserAnimelist.prototype.plantowatch = function (qparams) {
        return this.getList(["animelist", "plantowatch", this.page], qparams);
    };
    return UserAnimelist;
}());
var UserMangalist = /** @class */ (function () {
    function UserMangalist(parent, p) {
        this.parent = parent;
        if (p != null)
            this.page = String(p);
        else
            this.page = "1";
    }
    UserMangalist.prototype.getList = function (params, qparams) {
        if (qparams != null) {
            var qparams_2 = Object.keys(params)
                .filter(function (k) { return params[k] != null; })
                .map(function (k) { return k + "=" + encodeURIComponent(params[k]); })
                .join("&");
            params.push(qparams_2);
        }
        // @ts-ignore
        var url = url_1.joinUrl(this.parent.baseUrl, params);
        // @ts-ignore
        return this.parent.jikanGet(url);
    };
    UserMangalist.prototype.all = function (qparams) {
        return this.getList(["mangalist", "all", this.page], qparams);
    };
    UserMangalist.prototype.reading = function (qparams) {
        return this.getList(["mangalist", "reading", this.page], qparams);
    };
    UserMangalist.prototype.onhold = function (qparams) {
        return this.getList(["mangalist", "onhold", this.page], qparams);
    };
    UserMangalist.prototype.dropped = function (qparams) {
        return this.getList(["mangalist", "dropped", this.page], qparams);
    };
    UserMangalist.prototype.plantoread = function (qparams) {
        return this.getList(["mangalist", "plantoread", this.page], qparams);
    };
    return UserMangalist;
}());
function default_1(username) {
    return new User(username);
}
exports.default = default_1;
